src/pages/Sketch/components/Canvas.jsx

```diff
@@ -493,27 +498,45 @@ function Canvas() {
       });
   };
 
-  const initiateLoadSaved = () => document.getElementById("file-selector").click();
-  const loadLastState = e => {
+
+  
+ 
+  const initiateload = () => document.getElementById("file-selector").click();
+  const loadimport= e =>{   
     let file = e.target.files[0];
-    if (!file) return;
-    let reader = new FileReader();
+    var textFile = /image.*/;
+
+
+    if (!(file.type.match(textFile))) { 
+     alert("OOPs! You didn't select an image. We only support jpg,png and webp");
+        file.value = '';
+        return false;
+            }
+            else
+            {
+    let reader =  new FileReader();
     reader.addEventListener(
       "load",
       () => {
         const image = new Image();
 
-        image.onload = () => {
+        image.onload = () => { 
           context.drawImage(image, 0, 0);
-          setCanvasStates(current => [...canvasStates, context.getImageData(0, 0, canvasWidth, canvasHeight)]);
+          setCanvasStates(current => [...canvasStates, context.getImageData(0 , 0, canvasWidth, canvasHeight)]);
           setcanvasStateAt(current => current + 1);
         };
         image.src = reader.result;
       },
       false
     );
     reader.readAsDataURL(file);
-  };
+    
+  }; 
+};
+
+    
+    
+
```

Comment:
unformatted codes everywhere

Reply:
@ashuvssut since i m new to contribution , there aremistakes from my side in and the mentor has already pin pointed it before to which i have replied too. i will keep this in mind from next time , i have already told the mentor this thing. And i don't think u can just label it as a bad contribution since only the file format has been changed that too for making it more easy , its not that any feature isnt working

Reply:
>  its not that any feature isnt working

se my comment below. the gif clearly shows that your deleted existing good code and replaced with you code blindly